File: app/src/main/java/com/prey/services/CheckLockActivated.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2020 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.IBinder;

import com.prey.PreyConfig;
import com.prey.PreyLogger;
import com.prey.activities.LockScreenActivity;
import com.prey.json.actions.Lock;

public class CheckLockActivated extends Service {

    public IBinder onBind(Intent intent) {
        return null;
    }

    public void onCreate() {
        super.onCreate();
    }

    public void onStart(Intent intent, int startId) {
        super.onStart(intent, startId);
        final Context ctx = this;
        new Thread() {
            public void run() {
                boolean run = true;
                while (run) {
                    String unlockPass = PreyConfig.getPreyConfig(getApplicationContext()).getUnlockPass();
                    if (unlockPass == null || "".equals(unlockPass)) {
                        run = false;
                        stopSelf();
                        break;
                    }
                    try {
                        Thread.sleep(2000);
                        // Re-launch LockScreenActivity if it's not in foreground
                        Intent lockIntent = new Intent(ctx, LockScreenActivity.class);
                        lockIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                        ctx.startActivity(lockIntent);
                    } catch (Exception e) {
                        PreyLogger.e("CheckLockActivated Error:" + e.getMessage(), e);
                    }
                }
            }
        }.start();
    }

}

File: app/src/main/java/com/prey/services/kotlin/CheckLockActivated.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.services.kotlin

import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.IBinder
import com.prey.json.actions.kotlin.Lock
import com.prey.kotlin.PreyConfig
import com.prey.kotlin.PreyLogger

class CheckLockActivated : Service() {
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onCreate() {
        super.onCreate()
    }

    override fun onStart(intent: Intent, startId: Int) {
        super.onStart(intent, startId)
        val ctx: Context = this
        object : Thread() {
            override fun run() {
                var run = true
                while (run) {
                    val unlockPass = PreyConfig.getInstance(
                        applicationContext
                    ).getUnlockPass()
                    if (unlockPass == null || "" == unlockPass) {
                        run = false
                        stopSelf()
                        break
                    }
                    try {
                        sleep(1000)
                        if (!Lock.canDrawOverlays(ctx)) {
                            Lock.lockWhenYouNocantDrawOverlays(ctx)
                            stopSelf()
                            break
                        }
                    } catch (e: Exception) {
                        PreyLogger.e("CheckLockActivated Error:" + e.message, e)
                    }
                }
            }
        }.start()
    }
}